app/src/main/java/com/jshvarts/todoapp/arch/MviViewModel.kt

```diff
@@ -32,4 +23,8 @@ abstract class MviViewModel<A : UiAction, S : UiState, E : UiEffect>(
   fun dispatchAction(action: A) {
     return if (isStateInBundle) Unit else handleAction(action)
   }
+
+  interface HasUiEffect<E : UiEffect> {
```

Comment:
How about `EffectSource`?

Reply:
how about `EffectProducer` or `ProducesEffect`?

Reply:
nit: to me `source` implies that making effects is it's primary function which is not

Reply:
I like `EffectProducer`